firedrake/pointquery_utils.py

```diff
@@ -192,37 +193,46 @@ def predicate(index):
 
 
 @PETSc.Log.EventDecorator()
-def compile_coordinate_element(ufl_coordinate_element, contains_eps, parameters=None):
+def compile_coordinate_element(mesh, contains_eps, parameters=None):
```

Comment:
```suggestion
def compile_coordinate_element(mesh: MeshGeometry, contains_eps: float, parameters: dict | None = None):
```
Note `MeshGeometry` not `MeshTopology` (I might be wrong though).